src/klein/test/test_resource.py

```diff
@@ -988,7 +1008,7 @@ def test_url_for(self) -> None:
         app = self.app
         request = requestMock(b"/foo/1")
 
-        relative_url = [None]
+        relative_url: List[Optional[str]] = [None]
```

Comment:
Maybe use a helper string as the default value.
I hope that this should get better error messages if the test fails, you you don't need to make the type optional. 

```suggestion
        relative_url: List[str] = ['route-not-called']
```